### ISSUE TITLE
[Bug]: Search failed with error `fail to get shard leaders from QueryCoord: failed to get channels, err=CollectionNotFound`

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#21093]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20221208-eb7ef01b
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus==2.3.0.dev15
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2022-12-08T10:15:52.201Z] [2022-12-08 10:15:11 - DEBUG - ci_test]: (api_request)  : [Collection.delete] args: ['int64 in [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]', None, 120], kwargs: {} (api_request.py:56)

[2022-12-08T10:15:52.201Z] [2022-12-08 10:15:11 - DEBUG - ci_test]: (api_response) : (insert count: 0, delete count: 10, upsert count: 0, timestamp: 437910113311784962, success count: 0, err count: 0)  (api_request.py:31)

[2022-12-08T10:15:52.201Z] [2022-12-08 10:15:11 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[b'\xda\xb7pY\xe5\x9b\x90\xc3\x84\x1d\xc1m\xbf\x8eN\x1e', b'\xe2WMh\x9b\x0c\xc8\xfc\xc3f!\xb5yaV2'], 'binary_vector', {'metric_type': 'HAMMING', 'params': {'nprobe': 10}}, 10, 'int64 >= 0', None, None, 120, -1], kwargs: {} (api_request.py:56)

[2022-12-08T10:15:52.201Z] [2022-12-08 10:15:11 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=All attempts results:

[2022-12-08T10:15:52.201Z] attempt #1:fail to get shard leaders from QueryCoord: failed to get channels, err=CollectionNotFound

[2022-12-08T10:15:52.201Z] )>, <Time:{'RPC start': '2022-12-08 10:15:11.894034', 'RPC error': '2022-12-08 10:15:11.903700'}> (decorators.py:108)

[2022-12-08T10:15:52.201Z] [2022-12-08 10:15:11 - ERROR - ci_test]: Traceback (most recent call last):

[2022-12-08T10:15:52.201Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper

[2022-12-08T10:15:52.201Z]     res = func(*args, **_kwargs)

[2022-12-08T10:15:52.201Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request

[2022-12-08T10:15:52.201Z]     return func(*arg, **kwargs)

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 610, in search

[2022-12-08T10:15:52.201Z]     res = conn.search(self._name, data, anns_field, param, limit, expr,

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler

[2022-12-08T10:15:52.201Z]     raise e

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler

[2022-12-08T10:15:52.201Z]     return func(*args, **kwargs)

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler

[2022-12-08T10:15:52.201Z]     ret = func(self, *args, **kwargs)

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler

[2022-12-08T10:15:52.201Z]     raise e

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler

[2022-12-08T10:15:52.201Z]     return func(self, *args, **kwargs)

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 469, in search

[2022-12-08T10:15:52.201Z]     return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 438, in _execute_search_requests

[2022-12-08T10:15:52.201Z]     raise pre_err

[2022-12-08T10:15:52.201Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 429, in _execute_search_requests

[2022-12-08T10:15:52.201Z]     raise MilvusException(response.status.error_code, response.status.reason)

[2022-12-08T10:15:52.201Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=All attempts results:

[2022-12-08T10:15:52.201Z] attempt #1:fail to get shard leaders from QueryCoord: failed to get channels, err=CollectionNotFound

[2022-12-08T10:15:52.201Z] )>

[2022-12-08T10:15:52.201Z]  (api_request.py:39)

[2022-12-08T10:15:52.201Z] [2022-12-08 10:15:11 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=All attempts results:

[2022-12-08T10:15:52.201Z] attempt #1:fail to get shard leaders from QueryCoord: failed to get channels, err=CollectionNotFound

[2022-12-08T10:15:52.201Z] )> (api_request.py:40)
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_kafka_cron/detail/deploy_test_kafka_cron/66/pipeline

log:
[artifacts-kafka-cluster-reinstall-66-server-second-deployment-logs.tar.gz](https://github.com/milvus-io/milvus/files/10191710/artifacts-kafka-cluster-reinstall-66-server-second-deployment-logs.tar.gz)

[artifacts-kafka-cluster-reinstall-66-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/10191711/artifacts-kafka-cluster-reinstall-66-pytest-logs.tar.gz)

**Comment by yah01** [comment]
fixed with #21069 
/assign @zhuwenxing

**Comment by zhuwenxing** [comment]
Not reproduced in master-20221212-e977e014
